group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -874,4 +1021,34 @@ public void replay(
             consumerGroup.updateMember(newMember);
         }
     }
+
+    /**
+     * A new metadata image is available.
+     *
+     * @param newImage  The new metadata image.
+     * @param delta     The delta image.
+     */
+    public void onNewMetadataImage(MetadataImage newImage, MetadataDelta delta) {
+        metadataImage = newImage;
+
+        // Notify all the groups subscribed to the created, updated or
```

Comment:
Is created topics a different method in topicsDelta? Shouldn't we have `createdTopicIds` and we add them?

Reply:
I took a longer look at the code -- and assuming changedTopics includes the newly created topics, this should be fine.

Reply:
I guess my question then is what is the flow for updating the groups with the image? This will just happen on the next heartbeat since we set metadataImage to new image? 

We don't really do any notifying besides resetting the refresh timer.

Reply:
> Is created topics a different method in topicsDelta? Shouldn't we have createdTopicIds and we add them? Or is changedTopics accounting for that? Are there other changes besides topic creation we can have?

My understanding is that created topics are included in the changed topics as well.

> I guess my question then is what is the flow for updating the groups with the image? This will just happen on the next heartbeat since we set metadataImage to new image?

Right. The idea is to flag all the groups subscribed topics and to let them update themselves on the next heartbeat. I did it this way because we can also rely on this mechanism to refresh regex based subs every X minutes later on.